src/main/js/dynamic/loaded/users/RoleTable.js

```diff
@@ -80,136 +49,189 @@ class RoleTable extends Component {
+        this.setState({
+            role: newRole
+        });
+    }
+
+    createColumns() {
+        return [
+            {
+                header: 'id',
+                headerLabel: 'Id',
+                isKey: true,
+                hidden: true
+            },
+            {
+                header: 'roleName',
+                headerLabel: 'Name',
+                isKey: false,
+                hidden: false
+            }
+        ];
+    }
+
+    retrieveData() {
+        const { getRoles } = this.props;
+        getRoles();
     }
 
     async savePermissions(permission) {
         const { role, incrementalId } = this.state;
         const { permissions } = role;
-
-        if (!permission.id) {
-            permission.id = incrementalId;
+        const permissionCopy = JSON.parse(JSON.stringify(permission));
+        if (!permissionCopy.id) {
+            permissionCopy.id = incrementalId;
             this.setState({
                 incrementalId: incrementalId + 1
             });
-            permissions.push(permission);
+            permissions.push(permissionCopy);
         } else {
-            const matchingPermissionIndex = permissions.findIndex(listPermission => listPermission.id === permission.id);
+            const matchingPermissionIndex = permissions.findIndex((listPermission) => listPermission.id === permission.id);
             if (matchingPermissionIndex > -1) {
                 permissions[matchingPermissionIndex] = permission;
             }
         }
         role.permissions = permissions;
         this.setState({
-            role: role
+            role
         });
         return true;
     }
 
     deletePermission(permissionIds) {
         const { role } = this.state;
         const { permissions } = role;
-        const filteredPermissions = permissions.filter(listPermission => !permissionIds.includes(listPermission.id));
-        let newRole = { ...role };
+        const filteredPermissions = permissions.filter((listPermission) => !permissionIds.includes(listPermission.id));
+        const newRole = { ...role };
         newRole.permissions = filteredPermissions;
         this.setState({
             role: newRole
         });
     }
 
     createModalFields() {
-        const { role } = this.state;
+        const { role, incrementalId } = this.state;
 
         const { permissions } = role;
-        let incrementedId = this.state.incrementalId;
-        permissions.forEach(permission => {
-            if (!permission.id) {
-                permission.id = incrementedId;
-                incrementedId++;
+        let incrementedId = incrementalId;
+        const updatedPermissions = [];
+        permissions.forEach((permission) => {
+            const permissionCopy = JSON.parse(JSON.stringify(permission));
+            if (!permissionCopy.id) {
+                permissionCopy.id = incrementedId;
+                incrementedId += 1;
```

Comment:
Any specific reason we changed to this type of increment?

Reply:
eslint rule does not like ++ it recommends in this case +=.  
Example:
increment x by y
`x = x + y` equivalent to `x += y` here in our case y = 1

Reply:
Gotcha. That's a bit surprising.